generator/lib/control.ml

```diff
@@ -207,6 +207,18 @@ module Volume(R: RPC) = struct
        "extra metadata associated by [set] is not copied."]
       (dbg @-> sr @-> key @-> returning volume errors)
 
+  let copy = R.declare "copy"
+      ["[copy sr volume dest_sr] creates a new volume as a writeable copy of ";
+       "[volume] in [new_sr]. [new_sr] may be the same as [sr] and the operation ";
+       "may be rejected if the volume management plugin cannot copy a volume ";
+       "between different SRs. It is expected that this operation is accelerated ";
+       "by the implementation and is more efficient than a blockwise replication ";
+       "through the local host. This operation should only be called if the plugin ";
+       "declares the VDI_COPY feature in the query response. If the operation is ";
+       "rejected then the caller will be expected to fall back to performing a ";
+       "blockwise copy." ]
+      (dbg @-> sr @-> key @-> sr @-> returning volume errors)
```

Comment:
Is this for copying between SRs of the same type (that use the same volume plugin)?

Reply:
most likely it only applies to SR from the same provider, and possibly only the same provider and the same backing store.

Reply:
There are options using technologies similar to ODX (https://docs.microsoft.com/en-us/windows-hardware/drivers/storage/offloaded-data-transfer) where you can do accelerated copy operations between different backing stores using the same technology with separate interconnects.

Reply:
The expectation is also that the offloaded/accelerated copy offered by this operation is likely to be a Copy-on-Write model with refcounting and extent mapping so that it can optimise the space utilisation as well as the time taken to create the copy.